### PR TITLE
BPK-1520 - Fix android icon alignment

### PR DESCRIPTION
**Pull request by matthewdavidson**
We discovered that our icons on android were not vertically centered in their bounding box for both small and large icons. Turns out there is an android specific prop `includeFontPadding` (see https://facebook.github.io/react-native/docs/text.html#style) that includes extra padding to account for ascenders and descenders. Since these are icons in a 24 / 16 dp bounding box this does not apply so we set it to false.

**Note:** The screenshot for iOS icons looks like it has changed as a result of this but in fact it's just because the screenshot on master was out of date. The first commit illustrates this.

Before | After
------------ | -------------
before | after

### CODE REVIEW

**Comment by k0nserv** [comment]
How does this look for the navigation bar back button? That was spot on when I made it. I'm concerned this could have broken that

**Comment by matthewdavidson** [comment]
@k0nserv the screenshots of all affected components have been updated as part of this PR. See https://github.com/Skyscanner/backpack/pull/677/files?short_path=6e2748d#diff-6e2748d79ff505a54d97512254987d92 in you specific example.

**Comment by k0nserv** [comment]
This does indeed break the alignments on Android.

image
*Note that the text baseline doesn't line up with the end of the arrow anymore*

Maybe we need an API to support opting in to the new behaviour?

File: native/packages/react-native-bpk-component-icon/src/BpkIcon.js [hunk 1]
```diff
@@ -32,12 +32,11 @@ const styles = StyleSheet.create({
   icon: {
     fontFamily: 'BpkIcon',
     fontSize: spacingLg,
-    lineHeight: spacingLg,
```
Comment (k0nserv) [inline native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: Do we not still need this to ensure we get the exact height we want or is that embedded in the font?

Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-icon/src/BpkIcon.js]: Please refer to the images in the desc of this PR. 

The dimensions are now 24 x 24 as expected as a result of removing the line height.

**Comment by shaundon** [comment]
@k0nserv James did say in a previous comment:

> All good from design... we know the alignment is off, but will accept this for now as a re-design of this component is needed.

So maybe the navigation bar misalignment is an acceptable trade-off. I appreciate how hard it was for you to align them in the first place though 😅

**Comment by k0nserv** [comment]
> So maybe the navigation bar misalignment is an acceptable trade-off. I appreciate how hard it was for you to align them in the first place though 😅

No it wasn't actually hard because our icons follow the material spec and the box was the right size. It lined up perfectly from the first go.

**Pull request merged**